server/routes/charts.js

```diff
@@ -126,5 +144,49 @@ router
       return res.status(500).json({ message: error.message })
     }
   })
+  .get(ensureAuthenticated, async (req, res) => {
+    try {
+      const { dataDb } = req.app.locals
+      const chartForm = await dataDb
+        .collection(collections.charts)
+        .findOne({ _id: new ObjectID(req.params.chart_id) })
+      return res.status(200).json({ data: chartForm })
+    } catch (error) {
+      console.error(error)
 
+      return res.status(500).json({ message: error.message })
+    }
+  })
+  .put(ensureAuthenticated, async (req, res) => {
+    try {
+      const { dataDb } = req.app.locals
+      const { chart_id } = req.params
+      const { title, variable, assessment, description, fieldLabelValueMap } =
+        req.body
+      const { value, ok } = await dataDb
+        .collection(collections.charts)
+        .findOneAndUpdate(
+          { _id: new ObjectID(chart_id) },
+          {
+            $set: {
+              title,
+              variable,
+              assessment,
+              description,
+              fieldLabelValueMap,
+              updatedAt: new Date().toISOString(),
+            },
+          },
+          {
+            upsert: true,
+          }
+        )
+
+      return ok === 1
+        ? res.status(200).json({ data: value })
+        : res.status(400).json({ message: 'Chart information not found' })
```

Comment:
```suggestion
        : res.status(404).json({ message: 'Chart not found' })
```

Reply:
🤔 huh, I've always thought of `404` as a resource (page/endpoint) not being found, vs. a document/record not being found. In this case I think `400` makes sense, as the `id` in the request was invalid. What do y'all think?

Reply:
I think the page cannot be found because the resource cannot be found. For instance, here's how Github handles a repository that cannot be found.

<img width="1075" alt="Screen Shot 2022-08-02 at 12 56 41 PM" src="https://user-images.githubusercontent.com/2905145/182431787-6eae9119-1c5a-4d03-b1b4-bc4f3100fb91.png">

Reply:
ah yeah, where `foobarbaz` would be the `id` that makes sense 👍